src/main/java/org/example/persistence/PersistenceProgram.java

```diff
@@ -1,54 +1,422 @@
+            //connection.commit();
+            return true;
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return false;
+        }
+
+    }
+
 
     @Override
     public boolean deleteProgram(ProgramEntity programEntity) {
-        return false;
+        softDeleteProgramTable(programEntity);
+
+
+        return true;
+
     }
 
     @Override
     public boolean updateProgram(ProgramEntity programEntity) {
-        return false;
+        try {
+            PreparedStatement stmt = connection.prepareStatement("UPDATE  programinformation SET title = ?, description = ? " +
+                    " WHERE programinformation.id = ? ");
+            stmt.setString(1,programEntity.getName());
+            stmt.setString(2,programEntity.getDescription());
+            stmt.setLong(3,programEntity.getPrograminformationId());
+            stmt.executeUpdate();
+
+            return true;
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return false;
+        }
+
+
+    }
+
+    public ArrayList<CreditEntity> getCreditUser(ProgramEntity programEntity){
+        try {
+            PreparedStatement statement = connection.prepareStatement("SELECT credit.program, \"user\".id, \"user\".title, \"user\"" +
+                    ".firstname, \"user\".middlename ,\"user\".lastname,\"user\".email from credit inner join " +
+                    "\"user\" on credit.\"user\" = \"user\".id where credit.program = ? ");
+            statement.setLong(1,programEntity.getId());
+            ResultSet resultSet = statement.executeQuery();
+            ArrayList<CreditEntity> creditList = new ArrayList<>();
+            while (resultSet.next()){
+               CreditEntity credit = new CreditEntity(programEntity.getId(),createUserEntityFromResultSet(resultSet));
+               creditList.add(credit);
+            }
+            return creditList;
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public ArrayList<UserEntity> getProducerforProgram(ProgramEntity programEntity){
+        try {
+            PreparedStatement statement = connection.prepareStatement("select \"user\".id, \"user\".title,firstname, middlename, lastname, " +
+                    "email from programproducer inner join \"user\" on programproducer.producer_id = \"user\".id " +
+                    "where programproducer.program_id = ?");
+            statement.setLong(1,programEntity.getId());
+            ResultSet resultSet = statement.executeQuery();
+            ArrayList<UserEntity> producerList = new ArrayList<>();
+            while (resultSet.next()){
+            UserEntity producer = createUserEntityFromResultSet(resultSet);
+            producerList.add(producer);
+            }
+            return producerList;
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+
```

Comment:
Hvorfor beder du om et helt object, når du kun bruger id'et for det? hvad er tanken bag det?